lib/services/display_features_helper.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:ui';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class DisplayFeaturesChangeNotifier extends ChangeNotifier
+    with WidgetsBindingObserver {
+  BuildContext? _context;
+  List<DisplayFeature> _features = [];
+  bool _halfOpened = false;
+
+  List<DisplayFeature> get features => _features;
+
+  bool get halfOpened => _halfOpened;
+
+  void attach(BuildContext context) {
+    _context = context;
+    WidgetsBinding.instance.addObserver(this);
+    _update();
+  }
+
+  void detach() {
+    WidgetsBinding.instance.removeObserver(this);
+    _context = null;
+  }
+
+  void _update() {
+    final context = _context;
+    if (context != null) {
+      _features = View.of(context).displayFeatures;
+      _halfOpened = _features.any((d) =>
+          (d.type == DisplayFeatureType.fold ||
+              d.type == DisplayFeatureType.hinge) &&
+          // Flip-style foldable, top == bottom, height == 0
+          d.bounds.height == 0 &&
+          d.state == DisplayFeatureState.postureHalfOpened);
+      notifyListeners();
+    }
+  }
+
+  @override
+  void didChangeMetrics() => _update();
+}
+
+final displayFeaturesNotifierProvider =
+    ChangeNotifierProvider<DisplayFeaturesChangeNotifier>(
+  (ref) {
+    final notifier = DisplayFeaturesChangeNotifier();
+    ref.onDispose(notifier.detach);
+    return notifier;
+  },
+);
```

Comment:
@Komodo5197 this is probably not ideal again, could you tell me where I could put this instead? And is using such a provider even good practice?

Reply:
I have no idea, I'll have a look at the docs for that too!

Reply:
Not sure that will work with my ChangeNotifier. I'll keep it as-is for now.